### ISSUE TITLE
Classic scores show up in Selected Mods leaderboard even if NoMod is enabled

### ISSUE THREAD

**Issue by Non-Binary-Programmer** [ppy/osu#21935]
### Type

Game behaviour

### Bug description

If you enable Selected Mods and have no mods selected, scores set with Classic and no other mods (besides +SD and +PF) will show up on the leaderboard.

### Screenshots or videos

Example including a +PF score. Notice the lack of mods selected.
<img width="1512" alt="Screen Shot 2022-12-29 at 3 47 21 PM" src="https://user-images.githubusercontent.com/100527514/210014088-2b5222c1-486a-4dbd-8c7a-8cbdb0312fa5.png">

### Version

2022.1228.0-lazer

### Logs

[database.log](https://github.com/ppy/osu/files/10322237/database.log)
[input.log](https://github.com/ppy/osu/files/10322238/input.log)
[network.log](https://github.com/ppy/osu/files/10322239/network.log)
[performance-draw.log](https://github.com/ppy/osu/files/10322240/performance-draw.log)
[performance-update.log](https://github.com/ppy/osu/files/10322241/performance-update.log)
[performance.log](https://github.com/ppy/osu/files/10322242/performance.log)
[runtime.log](https://github.com/ppy/osu/files/10322243/runtime.log)

**Comment by peppy** [comment]
We definitely need to consider how classic mod plays into the systems and balance overall. This was one of the main discussion points I want to bring up with the community as we work towards a sane balance.

**Comment by peppy** [comment]
@nanaya to confirm here, this is an osu-web side consideration right now yeah? classic scores are included in search results even if the classic mod is not specified in a mods specific filter?

**Comment by nanaya** [comment]
yeah, it's based on `es_enable_legacy_filter` option whether or not to ignore CL mod

**Comment by peppy** [comment]
Thanks for the confirmation. I see this as an issue we can't immediately act on (ie. it doesn't have an objectively correct fix) without having a discussion about how we are moving forward with classic vs lazer scores, so I'll drop the priority of this for now.

**Comment by Walavouchey** [comment]
linking https://github.com/ppy/osu/discussions/27834 as related. the local scores tab *doesn't* ignore classic mod when filtering by selected mods, but i suppose whether that discrepancy should be fixed depends on the outcome of this issue
